client/services/code_preprocessors/PythonCodePreprocessorService.js

```diff
@@ -126,11 +129,32 @@ tie.factory('PythonCodePreprocessorService', [
         while (true) {
           var codeToAdd = '';
           lastFunctionNameLocation = code.indexOf(
-            functionName, lastFunctionNameLocation);
+            functionNameWithoutParen, lastFunctionNameLocation);
           if (lastFunctionNameLocation === -1) {
             break;
           }
           if (lastFunctionNameLocation !== functionNameStart) {
+            var whitespaceCheckLocation = (
+              lastFunctionNameLocation + functionNameWithoutParen.length);
+            // We assume we found the matched function, but we might have found
+            // a similarly-named other function / variable. User-inputted code
+            // can be ... tricky.
+            var isMatchedFunction = true;
+            // We can safely assume there's a left paren because we matched it
+            // with the regular expression.
+            while (code[whitespaceCheckLocation] != '(') {
+              var isValidWhitespace = VALID_WHITESPACE_REGEX.test(
```

Comment:
I did -- we have a test for functions that are similarly-named, now.

And yes, essentially -- we consistently move up lastFunctionNameLocation at the end of the while loop (or any time before we continue), so when we search for the index we _start_ from the new lastFunctionNameLocation such that we're always looking forward in the code string. Eventually, we will either hit no more matches (and return -1) and / or run out of code (and return -1), so we'll break.